zxlive/proof_panel.py

```diff
@@ -128,5 +167,87 @@ def _unfuse(self, v: VT, left_neighbours: list[VT]) -> None:
             new_g.remove_edge((v, neighbor))
         new_g.add_edge((v, left_vert))
         anim = anims.unfuse(self.graph, new_g, v, self.graph_scene)
-        cmd = SetGraph(self.graph_view, new_g)
+        cmd = AddRewriteStep(self.graph_view, new_g, self.step_view, "unfuse")
         self.undo_stack.push(cmd, anim_after=anim)
+
+    def _proof_step_selected(self, selected: QItemSelection, deselected: QItemSelection) -> None:
+        if not selected or not deselected:
+            return
+        cmd = GoToRewriteStep(self.graph_view, self.step_view, deselected.first().topLeft().row(), selected.first().topLeft().row())
+        self.undo_stack.push(cmd)
+
+
+class ProofStepItemDelegate(QStyledItemDelegate):
+    """This class controls the painting of items in the proof steps list view.
+
+    We paint a "git-style" line with circles to denote individual steps in a proof.
+    """
+
+    line_width = 3
+    line_padding = 13
+    vert_padding = 10
+
+    circle_radius = 4
+    circle_radius_selected = 6
+    circle_outline_width = 3
+
+    def paint(self, painter: QPainter, option: QStyleOptionViewItem, index: Union[QModelIndex, QPersistentModelIndex]) -> None:
+        painter.save()
+
+        # Draw background
+        painter.setPen(Qt.GlobalColor.transparent)
+        if option.state & QStyle.State_Selected:
+            painter.setBrush(QColor(204, 232, 255))
+        # TODO: Why is hover not working?
```

Comment:
It looks like it is working fine for me?

Reply:
I think this might be a MacOS thing, but I found a fix. We have to call `setAttribute(Qt.WA_Hover)` on the viewport of the list view. Now it also works on my machine 👍